vnext/Microsoft.ReactNative/ReactHost/CrashManager.cpp

```diff
@@ -4,106 +4,120 @@
+  // This allows the Windows Error Reporting system to process the error and upload the data for processing.
+  static LPTOP_LEVEL_EXCEPTION_FILTER g_previousExceptionFilter = nullptr;
 
-static std::wstring g_logFileName;
-static bool g_WERExceptionFilterWasCalled{false}; // Prevent recursion in the custom handler
+  static std::wstring g_logFileName;
+  static bool g_WERExceptionFilterWasCalled{ false }; // Prevent recursion in the custom handler
 
-/*static*/ std::atomic_uint CrashManager::_regCount{0};
+  /*static*/ std::atomic_uint CrashManager::_regCount{0};
 
-extern "C" LONG WINAPI CustomWERExceptionFilter(LPEXCEPTION_POINTERS const exceptionPointers) {
-  if (g_WERExceptionFilterWasCalled) {
-    return EXCEPTION_EXECUTE_HANDLER;
-  }
+  extern "C" LONG WINAPI CustomWERExceptionFilter(LPEXCEPTION_POINTERS const exceptionPointers) {
+    if (g_WERExceptionFilterWasCalled) {
+      return EXCEPTION_EXECUTE_HANDLER;
+    }
+
+    g_WERExceptionFilterWasCalled = true;
 
-  g_WERExceptionFilterWasCalled = true;
+    CrashManager::OnUnhandledException();
 
-  CrashManager::OnUnhandledException();
+    if (exceptionPointers == nullptr)
+    {
+      return EXCEPTION_EXECUTE_HANDLER;
+    }
 
-  return g_previousExceptionFilter(exceptionPointers);
-}
+    return g_previousExceptionFilter(exceptionPointers);
+  }
 
-void InternalRegisterCustomHandler() noexcept {
-  // Do this now because by the time we catch the exception we may be in OOM
+  void __cdecl on_sigabrt(int signum)
+  {
+    CustomWERExceptionFilter(nullptr);
+    signal(signum, SIG_DFL);
+  }
+
+  void InternalRegisterCustomHandler() noexcept {
+    // Do this now because by the time we catch the exception we may be in OOM
 #ifndef CORE_ABI // win32 vs uwp file permissions
-  wchar_t currentDirectory[MAX_PATH]{};
-  VerifyElseCrash(!!GetTempPath(MAX_PATH, currentDirectory));
-  g_logFileName = currentDirectory;
+    wchar_t currentDirectory[MAX_PATH]{};
+    VerifyElseCrash(!!GetTempPath(MAX_PATH, currentDirectory));
+    g_logFileName = currentDirectory;
 #else
-  g_logFileName = winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path() + L"\\";
+    g_logFileName = winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path() + L"\\";
 #endif
 
-  g_logFileName += L"ReactNativeCrashDetails_" + std::to_wstring(GetCurrentProcessId()) + L".txt";
+    g_logFileName += L"ReactNativeCrashDetails_" + std::to_wstring(GetCurrentProcessId()) + L".txt";
+
+    VerifySucceededElseCrash(::WerRegisterFile(g_logFileName.c_str(), WerRegFileTypeOther, WER_FILE_ANONYMOUS_DATA));
 
-  VerifySucceededElseCrash(::WerRegisterFile(g_logFileName.c_str(), WerRegFileTypeOther, WER_FILE_ANONYMOUS_DATA));
+    g_previousExceptionFilter = ::SetUnhandledExceptionFilter(CustomWERExceptionFilter);
 
-  g_previousExceptionFilter = ::SetUnhandledExceptionFilter(CustomWERExceptionFilter);
-}
+    signal(SIGABRT, &on_sigabrt);
```

Comment:
>signal(SIGABRT, &on_sigabrt); [](http://example.com/codeflow?start=4&length=29)

Can this end up overriding an existing handler, say if used in a brownfiels application like Office .. 
Can we chain handlers ?

Reply:
It'll be important to make sure folks can turn this behavior off, since this is the library dictating app behavior. For brownfield apps, that is likely not acceptable. I'm a little worried about this being applied kind of blanket-y. @jonthysell can you validate that this works for our other 1P partners?

Reply:
As mentioned in the description, this code doesn't run unless the app opts-in to default crash handling via setting EnableDefaultCrashHandler to true.